src/test/java/com/rackspace/salus/monitor_management/services/MonitorManagementTest.java

```diff
@@ -4378,4 +4378,71 @@ public void deleteAllMonitorsForTenant_noEvents() {
     assertThat(result.getNumberOfElements(), equalTo(0));
   }
 
+  @Test
+  public void testPatchExistingMonitor_nonExistingResource() {
+
+    final Monitor monitor = new Monitor()
+        .setAgentType(AgentType.TELEGRAF)
+        .setMonitorType(MonitorType.cpu)
+        .setContent("static content")
+        .setTenantId("t-1")
+        .setResourceId(null)
+        .setSelectorScope(ConfigSelectorScope.LOCAL)
+        .setLabelSelectorMethod(LabelSelectorMethod.AND)
+        .setLabelSelector(Map.of("os", "linux"))
+        .setZones(Collections.emptyList())
+        .setInterval(Duration.ofSeconds(60));
+    entityManager.persist(monitor);
+
+    // Called when binding new resource
+    when(resourceApi.getByResourceId("t-1", "r-1"))
+        .thenReturn(null);
+    when(resourceRepository.findByTenantIdAndResourceId("t-1", "r-1"))
+        .thenReturn(Optional.empty());
+
+    // Called when unbinding old resources
+    when(boundMonitorRepository.findResourceIdsBoundToMonitor(any()))
+        .thenReturn(Collections.emptySet());
+
+    // EXECUTE
+    String newResourceIdBinding = "r-1";
```

Comment:
```suggestion
    String newResourceIdBinding = RandomStringUtils.randomAlphanumeric(10);
```
One small suggestion.  Even though we have many tests that do it, I'd recommend not hardcoding values unless there's a specific reason for it.  We've seen issues in the past where setting things like `r-1` for resources all the time has hidden bugs that would have been exposed by randomizing the value instead.